### ISSUE TITLE
User input malformed: expected str for dictionary value @ data['brand_id']

### ISSUE THREAD

**Issue by McLP11** [fredericvl/py-agua-iot#42]
I have the following message after adding all data related to my Klover stove, installation went ok after importing the repository in HACS. Brand id I left 1 as default.

**Comment by Bimboboy007** [comment]
Gleiches Problem bei mir, komme auch nicht weiter.

**Comment by McLP11** [comment]
ok, thanks for sharing you are facing the same issue, so I'm not the only one...and no clue what I'm doing wrong...

**Comment by Bimboboy007** [comment]
<img width="401" alt="Bildschirmfoto 2022-12-18 um 23 09 12" src="https://user-images.githubusercontent.com/80967872/208322119-f62527f6-fecf-4fba-a545-42dede204d12.png">
<img width="1201" alt="Bildschirmfoto 2022-12-18 um 23 10 23" src="https://user-images.githubusercontent.com/80967872/208322120-5cc86142-2e02-4c0a-9b4a-113f7ad190cf.png">
<img width="1201" alt="Bildschirmfoto 2022-12-18 um 23 11 00" src="https://user-images.githubusercontent.com/80967872/208322125-fa7ebc50-2872-4678-8a82-0d70df207499.png">
Vielleicht hilft das bei der Fehlersuche.

**Comment by McLP11** [comment]
In your case I think the error is that you need to put the website address in the first box "Agua IOT URL" and nothing in the second box being required only if reported on the table.

**Comment by Bimboboy007** [comment]
Okay habe ich geändert sieht soweit ganz gut aus habe jetzt nur noch den Fehler Failed to login, please check credentials
obwohl bei der App die login Daten passen. Hab mich aus der App ausgeloggt und neu angemeldet da passen die Daten.

**Comment by bcornelissen1** [comment]
Hi, Same problem for me as  McLP11 , new user with a Piazzetta pellet stove.

![image](https://user-images.githubusercontent.com/116575630/208886600-2f5eb5ea-16bd-458e-bb13-abdcc61241c0.png)

**Comment by McLP11** [comment]
Ok so I'm not the only one, let's see if someone can find a solution...

**Comment by Bimboboy007** [comment]
So I have a basic question. Which prerequisites must be met in order to be able to integrate the oven at all? For me it's a Globefire Wiona and according to the manufacturer and from my own experience, it only works with the Globefire app and with no other app, i.e. it can't be controlled in any other way.
Tried it with the original app Nina definitely doesn't work.
But the login data is correct, I tested it with the Globefire app.
<img width="384" alt="Bildschirmfoto 2022-12-22 um 08 41 25" src="https://user-images.githubusercontent.com/80967872/209083009-80a08c9d-e2ba-4388-b412-41935938341c.png">

**Comment by GregZeMachine** [comment]
Hello

I managed to pass through this step today on my side after facing same issue.
Actually I just removed the 1 on the "Brand id" field, typed 1. And it worked.

If you don't put 1 you will have a "Failed to login". As mentioned in different readme, the 1 is mandatory. But I guess the default value written in the field was not the right type, so even though it looks the same from our perspective, it might not be the same data type. (anyway, that's just a guess on my side).

To sum up, just erase the 1 and type a fresh you one. It solved the issue on my side.

**Comment by bcornelissen1** [comment]
Thanks GregZeMachine,

Got a little further now i think, atleast no more error about the "brand id" , now im getting the following:

![image](https://user-images.githubusercontent.com/116575630/209354164-a464ac85-4cc0-418a-886f-e08d57a159d0.png)


![image](https://user-images.githubusercontent.com/116575630/209354141-953895fc-67a0-4e27-a077-669c5d4cc3e1.png)

**Comment by GregZeMachine** [comment]
Hello @bcornelissen1 
I can't really help for this one. 
Only thing I can share is that I oberserved this other issue where some fixes are proposed (maybe it is applicable for your situation) https://github.com/fredericvl/py-agua-iot/issues/8
Greg

**Comment by McLP11** [comment]
finally I was able to complete the integration process, removing and adding again the brand_id 1" as @GregZeMachine suggested, thanks!
the fact is now I can see only one of 2 stoves I have (Klover), and just one entity climate not active...need to work about

**Comment by Bimboboy007** [comment]
Have exactly the same problem, can add the device now but get no active entity.

**Comment by McLP11** [comment]
Finally after a restart I was able to see the climate sensor, but only of a single stove, not the second one I have, same model.
Those are the attributes of this climate sensor, trying to understand how to visualize and use the fan value for example.
hvac_modes: heat, off
min_temp: 7
max_temp: 35
target_temp_step: 0.5
fan_modes: 1, 2, 3, 4, 5
current_temperature: null
temperature: null
fan_mode: 3
hvac_action: heating
alarm_code: 0
device_status: 4
human_device_status: ON
real_power: 1
friendly_name: Sala
supported_features: 9

**Comment by SpeedBlack** [comment]
Hello !

I encounter the same problem. I can add the device now but get no active entity.
Do you find a solution ?

Many thanks !

**Comment by McLP11** [comment]
still have the issue, only one stove found and a single entity. This one works, I can switch change stove status and ventilation, so basically it's a climate entity. No other data collected...

**Comment by jipem01** [comment]
The library is compatible with multiples stoves. But the HA component was not. It's fixed with last release.
Since the python library and the HACS library have been splitted, make sure you update to the last version if the plugin.
You can find it here : https://github.com/vincentwolsink/home_assistant_micronova_agua_iot
